cli/cmd/status/status.go

```diff
@@ -217,24 +218,127 @@ func validEvent(events []release.HookEvent) bool {
 	return false
 }
 
-// checkConsulServers prints the status of Consul servers if they
-// are expected to be found in the Kubernetes cluster. It does not check for
-// server status if they are not running within the Kubernetes cluster.
-func (c *Command) checkConsulServers(namespace string) error {
+// checkConsulComponentsStatus fetch and prints the status of different consul components
+// like Consul Clients, Consul Servers, and Consul Deployments, in the given namespace of the cluster.
+func (c *Command) checkConsulComponentsStatus(namespace string) error {
+	var err error
+	var tbl *terminal.Table
+	tbl, err = c.getConsulClientsTable(namespace)
+	c.printComponentStatus(tbl, err, "Consul Clients")
+	tbl, err = c.getConsulServersTable(namespace)
+	c.printComponentStatus(tbl, err, "Consul Servers")
+	tbl, err = c.getConsulDeploymentsTable(namespace)
+	c.printComponentStatus(tbl, err, "Consul Deployments")
+	return err
+}
+
+// printComponentStatus prints the status of a given component (Consul Clients, Consul Servers, or Consul Deployments).
+func (c *Command) printComponentStatus(tbl *terminal.Table, err error, component string) {
+	c.UI.Output(fmt.Sprintf("%s status: ", component), terminal.WithHeaderStyle())
+	if err != nil {
+		c.UI.Output("unable to list %s: %s", component, err, terminal.WithErrorStyle())
+	}
+	if tbl != nil {
+		c.UI.Table(tbl)
+	} else {
+		c.UI.Output(fmt.Sprintf("No %s found in Kubernetes cluster.", component))
+	}
+}
+
+// getConsulClientsTable returns the table instance with the Consul Clients
+// and their ready status (number of pods ready/desired)
+func (c *Command) getConsulClientsTable(namespace string) (*terminal.Table, error) {
+	clients, err := c.kubernetes.AppsV1().DaemonSets(namespace).List(c.Ctx, metav1.ListOptions{LabelSelector: "app=consul,chart=consul-helm,component=client"})
+	if err != nil {
+		return nil, err
+	}
+	var tbl *terminal.Table
+	if len(clients.Items) != 0 {
+		tbl = terminal.NewTable("NAME", "READY", "AGE", "CONTAINERS", "IMAGES")
```

Comment:
FOR FUTURE : This can be constants scoped to the package so that they can be reused in tests. No need to hold the PR for this now.